src/mim/ast/emit.cpp

```diff
@@ -497,4 +497,13 @@ void CDecl::emit(Emitter& e) const {
     }
 }
 
+void RuleDecl::emit(Emitter& e) const {
+    auto _      = e.world().push(loc());
+    auto meta_t = e.world().rule_type(var()->emit_type(e));
+    auto rule_  = e.world().mut_rule(meta_t);
+    var()->emit_value(e, rule_->var());
+    const Rule* res = rule_->set(lhs()->emit(e), rhs()->emit(e), condition()->emit(e));
```

Comment:
order of eval for function calls is impl-defined in C++. This has the unfortunate effect that on some C++ impls we may experience slightly different stuff happening here. Better to split it like this:
```c++
auto l = lhs()->emit(e);
auto r = rhs()->emit(e);
auto c = condition()->emit(e);
auto res = rules_->set(l, r, c);
```